### ISSUE TITLE
Compiling fails with Apache 2.2.x

### ISSUE THREAD

**Issue by ghost** [bombadil/mod_proxy_msrpc#5]
Trying to compile on 32-bit Ubuntu Server 12.04.

Am I missing a package? Any help would be appreciated.

Wes

wjerrell@ubuntu-server-32:~/mod_proxy_msrpc$ apxs2 -c mod_proxy_msrpc.c
/usr/share/apr-1.0/build/libtool --silent --mode=compile --tag=disable-static i686-linux-gnu-gcc -prefer-pic -DLINUX=2 -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_REENTRANT -I/usr/include/apr-1.0 -I/usr/include/openssl -I/usr/include/xmltok -pthread     -I/usr/include/apache2  -I/usr/include/apr-1.0   -I/usr/include/apr-1.0   -c -o mod_proxy_msrpc.lo mod_proxy_msrpc.c && touch mod_proxy_msrpc.slo
mod_proxy_msrpc.c: In function 'proxy_msrpc_register_outlook_session':
mod_proxy_msrpc.c:143:31: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c:143:31: note: each undeclared identifier is reported only once for each function it appears in
mod_proxy_msrpc.c: In function 'proxy_msrpc_validate_outlook_session':
mod_proxy_msrpc.c:209:31: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_check_outlook_session_state':
mod_proxy_msrpc.c:277:31: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_process_data':
mod_proxy_msrpc.c:415:39: error: 'APLOG_TRACE1' undeclared (first use in this function)
mod_proxy_msrpc.c:429:43: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_tunnel':
mod_proxy_msrpc.c:537:35: warning: initialization makes pointer from integer without a cast [enabled by default]
mod_proxy_msrpc.c:539:35: warning: initialization makes pointer from integer without a cast [enabled by default]
mod_proxy_msrpc.c:551:31: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_read_and_parse_initial_pdu':
mod_proxy_msrpc.c:731:35: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c:793:31: error: 'APLOG_TRACE1' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_read_server_response':
mod_proxy_msrpc.c:921:31: error: 'APLOG_TRACE3' undeclared (first use in this function)
mod_proxy_msrpc.c:934:39: error: 'APLOG_TRACE1' undeclared (first use in this function)
mod_proxy_msrpc.c:1009:31: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_disconnect_backend':
mod_proxy_msrpc.c:1062:30: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c: In function 'proxy_msrpc_connect_backend':
mod_proxy_msrpc.c:1074:35: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c:1148:23: error: 'proxy_dir_conf' has no member named 'preserve_host'
mod_proxy_msrpc.c: In function 'proxy_msrpc_handler':
mod_proxy_msrpc.c:1281:31: error: 'APLOG_TRACE1' undeclared (first use in this function)
mod_proxy_msrpc.c:1291:61: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1303:65: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1308:61: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1311:35: error: 'APLOG_TRACE2' undeclared (first use in this function)
mod_proxy_msrpc.c:1317:73: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1360:69: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1365:61: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1371:66: error: 'proxy_worker_stat' has no member named 'ping_timeout_set'
mod_proxy_msrpc.c:1375:69: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1381:69: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1386:61: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1404:77: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1420:65: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1459:65: error: 'proxy_worker_stat' has no member named 'hostname'
mod_proxy_msrpc.c:1482:5: warning: passing argument 1 of 'apr_socket_close' makes pointer from integer without a cast [enabled by default]
/usr/include/apr-1.0/apr_network_io.h:307:27: note: expected 'struct apr_socket_t *' but argument is of type 'int'
apxs:Error: Command failed with rc=65536

**Comment by bombadil** [comment]
The source needs Apache httpd 2.4.x. Your build looks like you're using Apache httpd 2.2.x. Is that true?

**Comment by bombadil** [comment]
I've just added a configure check in commit 1d6ba33 that aborts configure with a short error message.

**Comment by AlexSamad** [comment]
Hi

Is there any word on 2.2 compatibility ?

**Comment by bombadil** [comment]
Hi Alex,

there is nothing specific to 2.4 in this module, so in principle it should be possible to change the source code to support Apache httpd 2.2.x too. Pull requests are welcome.

Micha Lenk

**Comment by AlexSamad** [comment]
Okay sounds like a challenge

Thanks at least i know I will not be wasting time

**Comment by dsever** [comment]
Hi, 

Any news about support for Apache  2.2?

Dubravko

**Comment by bombadil** [comment]
Hi,

sorry, no news regarding support for Apache 2.2.

Patches welcome though. :)

Micha

**Comment by jeffkaufman** [comment]
`APLOG_TRACE*` doesn't exist in Apache 2.2, so you could either remove these debugging messages or change them to `APLOG_DEBUG`

**Comment by bombadil** [comment]
Yes, but changing `APLOG_TRACE*` to `APLOG_DEBUG` wouldn't be enough. There were also some intrusive data structure changes between Apache 2.2 and 2.4.

**Comment by bombadil** [comment]
I'm sorry, but I have to admit I won't fix this issue anymore.

This project is obsolete for the following reason. Since Exchange 2016 the MSRPC protocol is deprecated by MAPI over HTTP, which seems to work just fine using the regular reverse proxy settings in Apache (i.e. mod_proxy_http).
